src/xhydro/indicators/pmp.py

```diff
@@ -671,3 +707,87 @@ def spatial_average_storm_configurations(da, *, radius):
         confi_ds.attrs["units"] = da.attrs["units"]
 
     return confi_ds
+
+
+def pw_snowfall(
+    pw: xr.DataArray,
+    method: str,
+    snow_events: xr.DataArray,
+    snw_threshold: float,
+    rainfall_events: xr.DataArray | None = None,
+    rf_threshold: float | None = None,
+    prec_events: xr.DataArray | None = None,
+):
+    """
+    Estimate precipitable water associated with snowfall events using various filtering methods, based on Klein et al. (2017).
+
+    Parameters
+    ----------
+    pw : xr.DataArray
+        DataArray containing the precipitable water.
+    method : {"m1", "m2", "m3"}
+        Method used to identify snowfall-associated precipitable water:
+        - m1:Selects time steps with at least `snw_threshold` snowfall and less than or equal to `rf_threshold` rainfall.
+        - m1:Selects time steps with snowfall greater than `snw_threshold`, regardless of rainfall.
+        - m3:Starts from m2 selection, but if rainfall exceeds `rf_threshold`, the precipitable water is scaled by
+        the ratio of snowfall to total precipitation (`snow_events / prec_events`).
+    snow_events : xr.DataArray
+        DataArray containing snowfall event amounts.
+    snw_threshold :  float
+        Minimum snowfall threshold used to filter events.
+    rainfall_events : xr.DataArray, optional
+        Required for methods "m1" and "m3". DataArray containing rainfall event amounts.
+    rf_threshold : float, optional
```

Comment:
Même chose. Peut-être aussi à changer ailleurs ?
```suggestion
    prra_threshold : float, optional
```